pkg/k8shandler/statefulset.go

```diff
@@ -504,6 +509,28 @@ func (node *statefulSetNode) isChanged() bool {
 
 		node.self.Spec.Template.Spec.Containers[index] = nodeContainer
 	}
-
 	return changed
 }
+
+func (node *statefulSetNode) progressUnshedulableNode(upgradeStatus *api.ElasticsearchNodeStatus) error {
+	if node.isChanged() {
+		if err := node.executeUpdate(); err != nil {
+			return err
+		}
+
+		partition, err := node.partition()
+		if err != nil {
+			return err
+		}
+
+		podName := fmt.Sprintf("%v-%v", node.name(), partition)
```

Comment:
isn't partition always going to be `0` here? or will it be whatever pod we last updated?